README.md

```diff
@@ -2,8 +2,69 @@
 Welcome!
 
 ---
-##### Third Party Notice
+## Third Party Notice
 This repository is not responsible for scripts malfunctioning due to issues that are not caused by asl-help.
 We reserve the right to immediately close any issues which report problems out of our control.
 
 Instead, please either open an issue in the repository of the script's creator, or ask a question in the [Speedrun Tool Development Discord server](https://discord.gg/cpYsxz7).
+
+## Purpose
+Asl-help was created to aid autosplitter authors in writing scripts for games that use the mono framework, specifically the Unity fork. It is packaged as a .NET framework library that can be loaded at runtime from an autosplitter script, and uses codegen to allow for the creation of more streamlined, human-readable, and update-resilient pointer paths.
```

Comment:
Please do not condense it down to "codegen", as an inexperienced user may not know what that means.